src/ol/layers/MapboxLayer.js

```diff
@@ -173,6 +191,34 @@ export default class MapboxLayer extends Layer {
     );
   }
 
+  /**
+   * Returns a style URL with apiKey & apiKeyName infos.
+   */
+  createStyleUrl() {
+    let style;
+    if (this.apiKey === false) {
+      style = this.styleUrl;
+    } else {
+      const [baseUrl, searchUrl] = this.styleUrl.split('?');
+      const parameters = qs.parse(searchUrl);
+      if (!this.apiKey) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `No apiKey is defined for MapboxLayer's request to ${this.styleUrl}`,
+        );
+        return null;
+      }
+      if (parameters[this.apiKeyName]) {
+        // replace key value from apiKey parameter.
+        parameters[this.apiKeyName] = this.apiKey;
+        style = `${baseUrl}?${qs.stringify(parameters)}`;
+      } else {
+        style = `${this.styleUrl}?${this.apiKeyName}=${this.apiKey}`;
+      }
+    }
+    return style;
+  }
```

Comment:
looks like there is a duplication of code.

Reply:
Centralized in a util fct 'getMapboxStyle'